cdap-unit-test/src/main/java/co/cask/cdap/test/internal/DefaultServiceManager.java

```diff
@@ -103,8 +112,23 @@ public URL getServiceURL() {
   @Override
   public URL getServiceURL(long timeout, TimeUnit timeoutUnit) {
     String discoveryName = String.format("service.%s.%s.%s", accountId, applicationId, serviceName);
-    String path = String.format("%s/namespaces/%s/apps/%s/services/%s/methods/",
+    ServiceDiscovered discovered = discoveryServiceClient.discover(discoveryName);
+    return createURL(new RandomEndpointStrategy(discovered).pick(timeout, timeoutUnit), applicationId, serviceName);
+  }
+
+  private URL createURL(@Nullable Discoverable discoverable, String applicationId, String serviceName) {
```

Comment:
I am not a big fan of util class. Ideally they can be extends from some abstract base class. However it requires bigger refactor on how the classes are organized. Probably do it in a separate PR as it's not the focus of this PR.